Update user and group models.

ttung opened the pull request:
### Description
This brings the user and group models in compliance with the schema in #30.

Depends on #30, #42

#### Issue
[ch64743](https://app.clubhouse.io/genepi/stories/space/64743)

### Test plan
run migration.

phoenixAja commented on src/py/aspen/database/models.py:
```diff
@@ -26,6 +26,8 @@ class Group(base):
 
     name = Column(String, unique=True, nullable=False)
+    gisaid_division_name = Column(String, unique=True)
+    email = Column(String, unique=True, nullable=False)
```
wouldn't a better way to do this be to query all users in a group and get the emails from each user record?

phoenixAja replied:
or i guess i don't see the use case of having a group email and user emails.

ttung replied:
So the reason is that public archives seem to want an email address.  In that case, whose email address should we use?  It would seem that a group email address is appropriate here.

phoenixAja replied:
ok makes sense.

phoenixAja reviewed (approved):
lg!

Commit:
Update user and group models.

### Description
This brings the user and group models in compliance with the schema in #30.

#### Issue
[ch64743](https://app.clubhouse.io/genepi/stories/space/64743)

### Test plan
run migration.
